app/main.py

```diff
@@ -1,2 +1,126 @@
+from typing import Any, Iterator, Hashable
+from dataclasses import dataclass
+
+
+@dataclass
+class Node:
+    key: Hashable
+    value: Any
+    hash_value: int | float
+    next_node: "Node" = None
+
+
 class Dictionary:
-    pass
+
+    def __init__(self) -> None:
+        self._length = 0
+        self._capacity = 8
+        self._hash_table = [None] * self._capacity
+
+    def _get_index_and_hash(self, key: Hashable) -> tuple:
+        keys_hash = hash(key)
+        index = keys_hash % self._capacity
+        return index, keys_hash
+
+    def _resize(self) -> None:
+        self._length = 0
+        self._capacity *= 2
+        old_hash_table = self._hash_table
+        self._hash_table = [None] * self._capacity
+
+        for node in old_hash_table:
+            while node is not None:
+                self[node.key] = node.value
+                node = node.next_node
+
+    def __getitem__(self, key: Hashable) -> None:
+        index, _ = self._get_index_and_hash(key)
+        current_node = self._hash_table[index]
+        while current_node is not None:
+            if current_node.key == key:
+                return current_node.value
+            current_node = current_node.next_node
+        raise KeyError(f"There are no key {key}")
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        index, keys_hash = self._get_index_and_hash(key)
+
+        if self._hash_table[index] is None:
+            self._hash_table[index] = Node(key, value, keys_hash)
+            self._length += 1
+            return
+
+        current = self._hash_table[index]
+        while current is not None:
+            if current.key == key:
+                current.value = value
+                return
+
+            if current.next_node is None:
+                break
+            current = current.next_node
+
+        current.next_node = Node(key, value, keys_hash)
+        self._length += 1
+
+        if self._length >= self._capacity * 2 / 3:
+            self._resize()
```

Comment:
This check should be at the beginning of the function

Reply:
Still not fixed